src/client.ts

```diff
@@ -425,14 +432,49 @@ in an environmental variable named FAUNA_SECRET or pass it to the Client\
       headerObject["x-last-txn-ts"] = this.#lastTxnTs;
     }
   }
+
+  #validateConfiguration() {
+    const config = this.#clientConfiguration;
+    if (config.client_timeout_ms) {
+      if (!config.query_timeout_ms) {
+        throw new TypeError(
+          `Expected number, but received ${config.query_timeout_ms}. 'query_timeout_ms' must be configured if 'client_timeout_ms' is configured.`
+        );
+      }
+
+      if (
+        config.client_timeout_ms <
+        config.query_timeout_ms + CLIENT_TIMEOUT_BUFFER
+      ) {
+        throw new RangeError(
+          `'client_timeout_ms' is less than minimum required setting. 'clien_timeout_ms' must be greater than or equal to 'query_timeout_ms' + ${CLIENT_TIMEOUT_BUFFER} ms`
+        );
+      }
+    }
+  }
+
+  #validateRequest(_: Query, queryOptions?: QueryRequestHeaders) {
+    if (
+      queryOptions?.query_timeout_ms &&
+      this.#clientConfiguration.client_timeout_ms &&
+      this.#clientConfiguration.client_timeout_ms <
+        queryOptions.query_timeout_ms + CLIENT_TIMEOUT_BUFFER
+    ) {
+      throw new RangeError(
+        `'query_timeout_ms' is more than maximum allowed setting. 'clien_timeout_ms' must be greater than or equal to 'query_timeout_ms' + ${CLIENT_TIMEOUT_BUFFER} ms`
```

Comment:
I suppose another option woudl be to let uers specify only a queryTImeout and a clientTimeoutBuffer 🤔 then they don't have to maintain two timeout settings.